lib/msal-node/src/config/ManagedIdentityId.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    ManagedIdentityErrorCodes,
+    createManagedIdentityError,
+} from "../error/ManagedIdentityError";
+import {
+    DEFAULT_MANAGED_IDENTITY_ID,
+    ManagedIdentityIdType,
+} from "../utils/Constants";
+import { ManagedIdentityIdParams } from "./Configuration";
+
+export class ManagedIdentityId {
+    private id: string;
+    private idType: ManagedIdentityIdType;
+    private isUserAssignedId: boolean;
+
+    constructor(managedIdentityIdParams?: ManagedIdentityIdParams) {
+        const userAssignedClientId =
+            managedIdentityIdParams?.userAssignedClientId;
+        const userAssignedResourceId =
+            managedIdentityIdParams?.userAssignedResourceId;
```

Comment:
Are these 3 options all implemented for IMDS and MSI? If they are, are you missing tests?

Reply:
I believe the implementation for each of them is one and the same. I need to write unit tests for each one though.